### ISSUE TITLE
False positive for 'chained-comparison'

### ISSUE THREAD

**Issue by skirpichev** [PyCQA/pylint#5800]
### Bug description

```
$ cat -n a.py 
     1  a = 1
     2  b = 2
     3  c = 3
     4  d = 4
     5  if a < 0 and b > 1 and c < 1 and d > 2:
     6      print(1)
```

### Configuration

```ini
None
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
************* Module a
a.py:1:0: C0114: Missing module docstring (missing-module-docstring)
a.py:1:0: C0103: Constant name "a" doesn't conform to UPPER_CASE naming style (invalid-name)
a.py:2:0: C0103: Constant name "b" doesn't conform to UPPER_CASE naming style (invalid-name)
a.py:3:0: C0103: Constant name "c" doesn't conform to UPPER_CASE naming style (invalid-name)
a.py:4:0: C0103: Constant name "d" doesn't conform to UPPER_CASE naming style (invalid-name)
a.py:5:3: R1716: Simplify chained comparison between the operands (chained-comparison)
```

### Expected behavior

No R1716.

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.3
Python 3.10.0+ (heads/3.10:045f205ba4, Nov  8 2021, 05:21:54) [GCC 10.2.1 20210110]
```

**Comment by mbyrnepr2** [comment]
It looks like this checker wants the following refactor:

Before:
`if a < 0 and b > 1 and c < 1 and d > 2:`

After:
`if a < 0 and b > 1 > c and d > 2:`


Some more details about the upper/lower bounds of `1`:
```
(Pdb) uses
defaultdict(<function RefactoringChecker._check_chained_comparison.<locals>.<lambda> at 0x106e54c10>, {'a': {'lower_bound': {<Compare l.6 at 0x106d3df00>}, 'upper_bound': set()}, 0: {'lower_bound': set(), 'upper_bound': {<Compare l.6 at 0x106d3df00>}}, 'b': {'lower_bound': set(), 'upper_bound': {<Compare l.6 at 0x106d3dd80>}}, 1: {'lower_bound': {<Compare l.6 at 0x106d3dd80>}, 'upper_bound': {<Compare l.6 at 0x106d3df30>}}, 'c': {'lower_bound': {<Compare l.6 at 0x106d3df30>}, 'upper_bound': set()}, 'd': {'lower_bound': set(), 'upper_bound': {<Compare l.6 at 0x106d3d1b0>}}, 2: {'lower_bound': {<Compare l.6 at 0x106d3d1b0>}, 'upper_bound': set()}})
(Pdb) uses['lower_bound']
{'lower_bound': set(), 'upper_bound': set()}
(Pdb) uses[1]['lower_bound']
{<Compare l.6 at 0x106d3dd80>}
(Pdb) [i.as_string() for i in uses[1]['lower_bound']]
['b > 1']
(Pdb) [i.as_string() for i in uses[1]['upper_bound']]
```

**Comment by mbyrnepr2** [comment]
Would it be an idea for Pylint to output the suggested refactor to the user?

**Comment by Pierre-Sassoulas** [comment]
Thank you @mbyrnepr2 I didn't see it and was ready to call that a false positive 😄 A warning message suggesting what to do would make a lot of sense.

**Comment by skirpichev** [comment]
@areveny, probably this could be closed in favor of #5814?
